### PR TITLE
CALLDATASIZE Gadget

### PR DESCRIPTION
**Pull request by roynalnaruto**
Following specs in [Specs PR](https://github.com/appliedzkp/zkevm-specs/pull/96)

### CODE REVIEW

**Comment by roynalnaruto** [comment]
Should make use of `StackOnlyOpcode::<0>::gen_associated_ops` once #73 is merged.

File: zkevm-circuits/src/evm_circuit/util/constraint_builder.rs [hunk 1]
```diff
@@ -436,7 +436,7 @@ impl<'a, F: FieldExt> ConstraintBuilder<'a, F> {
         word
     }
 
-    fn tx_context_lookup(
+    pub(crate) fn tx_context_lookup(
```
Comment (icemelon) [inline zkevm-circuits/src/evm_circuit/util/constraint_builder.rs:439]: revert this change as it is not required in this PR.

**Review by icemelon — DISMISSED** [review]
Otherwise LGTM

File: bus-mapping/src/evm/opcodes.rs [hunk 2]
```diff
@@ -100,7 +102,7 @@ fn fn_gen_associated_ops(opcode_id: &OpcodeId) -> FnGenAssociatedOps {
         // OpcodeId::CALLER => {},
         // OpcodeId::CALLVALUE => {},
         // OpcodeId::CALLDATALOAD => {},
-        // OpcodeId::CALLDATASIZE => {},
+        OpcodeId::CALLDATASIZE => CallDataSize::gen_associated_ops,
```
Comment (icemelon) [inline bus-mapping/src/evm/opcodes.rs:105]: You can now use `StackOnlyOpcode` as `TIMESTAMP` opcode is already merged.

**Review by icemelon — APPROVED** [review]
After fix the comment and rebase to latest main, I'll create the PR to upstream.

File: bus-mapping/src/evm/opcodes.rs (hunk 2 above)
Comment (roynalnaruto) [reply bus-mapping/src/evm/opcodes.rs:105]: @icemelon rebasing was a mess (because of the merge commit that got through). I'm closing this PR and opening a new one with all the changes from before as well as the changes suggested in the reviews.

**Comment by roynalnaruto** [comment]
Closing in favour of #90
